src/grisp_tools_io.erl

```diff
@@ -0,0 +1,97 @@
+-module(grisp_tools_io).
+
+-export([
+         ask/2,
+         ask/3
+        ]).
+
+ask(Prompt, Type) ->
+    ask_convert(Prompt, fun get/2, Type, none).
+
+ask(Prompt, Type, Default)  ->
+    ask_convert(Prompt, fun get/2, Type, Default).
+
+ask_convert(Prompt, TransFun, Type,  Default) ->
+    DefaultPrompt = erlang:iolist_to_binary([Prompt, default(Default), "> "]),
+    NewPrompt = erlang:binary_to_list(DefaultPrompt),
+    Data = trim(trim(io:get_line(NewPrompt)), both, [$\n]),
+    case TransFun(Type, Data)  of
+        no_data ->
+            maybe_continue(Prompt, TransFun, Type, Default);
+        no_clue ->
+            continue(Prompt, TransFun, Type, Default);
+        Ret ->
+            Ret
+    end.
+
+maybe_continue(Prompt, TransFun, Type, Default) ->
+    case Default of
+        none ->
+            continue(Prompt, TransFun, Type, Default);
+        Default ->
+            TransFun(Type, Default)
+    end.
+
+continue(Prompt, TransFun, Type, Default) ->
+    say("I didn't understand that. A ~p is expected.~n", [Type]),
```

Comment:
I am not a fan of machines pretending to be humans, but that is just me.
But maybe we should say to write y/yes or n/no
```
App name ("robot")> ttrtr
Erlang version ("25")> trtrt
Destination ("/path/to/SD-card")> ertrete
Network configuration(y/N)> etretet
I didn't understand that. A boolean is expected.
```